sdk/endpoints/online/managed/online-endpoints-managed-identity-sai.ipynb

```diff
@@ -0,0 +1,473 @@
+    "        data=f.read()\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create an online endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "create-endpoint-1"
+   },
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml import MLClient\n",
+    "ml_client = MLClient(credential, subscription_id, resource_group, workspace_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "create-endpoint-2"
+   },
+   "outputs": [],
+   "source": [
+    "sai_endpoint = ml_client.online_endpoints.begin_create_or_update(sai_endpoint)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "create-endpoint-3"
+   },
+   "outputs": [],
+   "source": [
+    "sai_endpoint.identity.as_dict()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+   },
+   "source": [
+    "## Give access permission to the managed identity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "give-access-permission-1"
+   },
+   "outputs": [],
+   "source": [
+    "from azure.mgmt.authorization import AuthorizationManagementClient\n",
+    "from azure.mgmt.authorization.models import RoleAssignment, RoleDefinition, RoleAssignmentCreateParameters, RoleAssignmentProperties, RoleAssignmentPropertiesWithScope\n",
+    "import uuid\n",
+    "\n",
+    "auth_client = AuthorizationManagementClient(\n",
+    "    credential=credential, \n",
+    "    subscription_id=subscription_id\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "give-access-permission-2"
+   },
+   "outputs": [],
+   "source": [
+    "sai_endpoint = ml_client.online_endpoints.get(sai_endpoint_name)\n",
+    "system_principal_id = sai_endpoint.identity.principal_id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "give-access-permission-3"
+   },
+   "outputs": [],
+   "source": [
+    "workspace = ml_client.workspaces.get(workspace_name)\n",
+    "storage_account = workspace.storage_account"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "name" : "give-access-permission-4"
+   },
+   "outputs": [],
+   "source": [
+    "role_name = \"Storage Blob Data Reader\"\n",
+    "scope = storage_account \n",
+    "\n",
+    "role_defs = auth_client.role_definitions.list(scope=scope)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "auth_client.role_assignments.create(\n",
+    "    scope=scope, \n",
+    "    role_assignment_name=uuid.uuid4(),\n",
+    "    parameters=RoleAssignmentProperties(\n",
+    "        role_definition_id = role_def.id, \n",
+    "        principal_id = system_principal_id\n",
+    "    )\n",
```

Comment:
I've faced the following error while running the code.
Did you face the same error, or do you have any ideas for this? Sorry..

```
HttpResponseError: (MalformedRoleAssignmentRequest) PASRP RoleAssignment request is malformed
Code: MalformedRoleAssignmentRequest
Message: PASRP RoleAssignment request is malformed
```

Reply:
Working on debugging this but has to be indirectly through the runner as I don't actually have role assignment permissions on any resource.

Reply:
I see. I'll give you permission if I can..